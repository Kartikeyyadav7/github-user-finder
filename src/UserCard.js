import React from "react";
import { addCommas } from "./helpers";
import { Link } from "@reach/router";

class UserCard extends React.Component {
  render() {
    const {
      title,
      username,
      id,
      avatarUrl,
      bio,
      location,
      hireable,
      stargazers,
      followers,
      repos
    } = this.props.user;
    return (
      <div className="item" id={username} key={id}>
        <div className="ui small image">
          <img src={avatarUrl} alt="User avatar" />
        </div>
        <div className="middle aligned content">
          <div className="header">
            {title}
            <br />
            {username}
          </div>
          <div className="description">
            <p>{bio}</p>
          </div>
          <div className="extra" style={{ color: "#A0A0A0" }}>
            <span>Location:</span>
            {location}
            <p style={{ color: "#FF1493" }}>
              {hireable ? "Availble for job offers" : null}
            </p>
          </div>
          <div className="extra content">
            <span>
              <i className="star icon" />
              {addCommas(stargazers)} Stargazers
            </span>
            <span>
              <i className="users icon" />
              {addCommas(followers)} Followers
            </span>
            <span>
              <i className="file code outline icon" />
              {addCommas(repos)} Repos
            </span>
            <div className="ui right floated">
              <Link to={`/details/${username}`}>
                <button className="ui primary button">More Info</button>
              </Link>
              <button
                className="ui button"
                onClick={() => this.props.handleDelete(username)}
              >
                Discard
              </button>
            </div>
          </div>
        </div>
      </div>
    );
  }
}

export default UserCard;
