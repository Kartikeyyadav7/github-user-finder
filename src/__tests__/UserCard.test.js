import React from "react";
import { create } from "react-test-renderer";
import UserCard from "../UserCard";

test("snapshot", () => {
  const user = {
    title: "String",
    username: "String",
    id: 0,
    avatarUrl: "string",
    bio: "string",
    location: "string",
    hireable: true,
    stargazers: 2000,
    followers: 2000,
    repos: 2000
  };
  const c = create(<UserCard user={user} />);
  expect(c.toJSON()).toMatchSnapshot();
});
