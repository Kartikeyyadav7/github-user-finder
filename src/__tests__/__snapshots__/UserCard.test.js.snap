// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<div
  className="item"
  id="String"
  style={
    Object {
      "backgroundColor": "#fafafa",
      "boxShadow": "0.2rem 0.2rem 0.2rem rgba(0,0,0,.35)",
      "margin": "1rem",
      "padding": "0.8rem",
      "width": "95%",
    }
  }
>
  <div
    className="ui small image"
  >
    <img
      alt="User avatar"
      src="string"
    />
  </div>
  <div
    className="middle aligned content"
  >
    <div
      className="header"
    >
      String
      <br />
      String
    </div>
    <div
      className="description"
    >
      <p>
        string
      </p>
    </div>
    <div
      className="extra"
      style={
        Object {
          "color": "#A0A0A0",
        }
      }
    >
      <span>
        Location:
      </span>
      string
      <p
        style={
          Object {
            "color": "#FF1493",
          }
        }
      >
        Availble for job offers
      </p>
    </div>
    <div
      className="extra content"
    >
      <span>
        <i
          className="star icon"
        />
        2,000
         Stargazers
      </span>
      <span>
        <i
          className="users icon"
        />
        2,000
         Followers
      </span>
      <span>
        <i
          className="file code outline icon"
        />
        2,000
         Repos
      </span>
      <div
        className="ui right floated"
      >
        <a
          href="/details/String"
          onClick={[Function]}
        >
          <button
            className="ui primary button"
          >
            More Info
          </button>
        </a>
        <button
          className="ui button"
          onClick={[Function]}
        >
          Discard
        </button>
      </div>
    </div>
  </div>
</div>
`;
